### PR TITLE
updating last_read_message and unread_count when marking a message as read

### PR DESCRIPTION
**Pull request by howawong**
Connects SkygearIO/chat-SDK-JS#22 and SkygearIO/chat-SDK-iOS#38

### CODE REVIEW

**Comment by oursky-travis** [comment]
@howawong, thanks for your PR! By analyzing the history of the files in this pull request, we identified @rickmak to be a potential reviewer.

File: chat/receipt_handlers.py [hunk 1]
```diff
@@ -63,6 +112,76 @@ def __fetch_receipts(messages, user_id):
...
+def __update_last_read_messages(last_read_messages, message, key):
+    last_read_message = last_read_messages.get(key, None)
+    if last_read_message is None:
+        last_read_message = message
+    if message.record['seq'] > last_read_message.record['seq']:
+        last_read_message = message
+    last_read_messages[key] = last_read_message
+
+
 def mark_messages(message_ids, mark_delivered, mark_read):
```
Comment (rickmak) [inline chat/receipt_handlers.py]: I am suggesting separate the `upsert receipts` into a separate function, something like `__upsert_receipts `. So the `mark_messages` will become smaller and only represent the sequence of work to be done. Not holding the actual work logic.

File: chat/receipt_handlers.py [hunk 2]
```diff
@@ -63,6 +112,76 @@ def __fetch_receipts(messages, user_id):
...
 
 
+def __update_user_conversations(unread_counts, last_read_messages, conn):
+    """
+    Update user conversation table based on un-read counts and
+    last read messages
+    """
+    user_id = current_user_id()
+    schema_name = AsIs(_get_schema_name())
+    for key in unread_counts:
```
Comment (rickmak) [inline chat/receipt_handlers.py]: `unread_counts` -> `unread_counter`?

I think give a hint annotation on telling `last_read_messages` is a dict will help other understands.

File: chat/receipt_handlers.py (hunk 1 above)
Comment (howawong) [inline chat/receipt_handlers.py]: extracted function __process_message_receipts().

File: chat/receipt_handlers.py (hunk 2 above)
Comment (howawong) [inline chat/receipt_handlers.py]: Updated.

**Comment by rickmak** [comment]
```
2017-07-12 18:34:00,497 ERROR [skygear.transmitter.common:57][Thread-30] Error occurred processing request
Traceback (most recent call last):
  File "/Users/rickmak/Working/py-skygear/skygear/transmitter/common.py", line 40, in wrapper
    return dict(result=f(self, *args, **kwargs))
  File "/Users/rickmak/Working/py-skygear/skygear/transmitter/common.py", line 100, in call_func
    return self.op(obj, param.get('args', {}))
  File "/Users/rickmak/Working/py-skygear/skygear/transmitter/common.py", line 174, in op
    return func(*args, **kwargs)
  File "chat/receipt_handlers.py", line 283, in mark_as_delivered_lambda
    return handle_mark_as_delivered(arg)
  File "chat/receipt_handlers.py", line 34, in handle_mark_as_delivered
    mark_messages(message_ids, True, False)
  File "chat/receipt_handlers.py", line 242, in mark_messages
    mark_read)
```

**Pull request merged**